app/jobs/dspace_publication_results_job.rb

```diff
@@ -0,0 +1,77 @@
+class DspacePublicationResultsJob < ActiveJob::Base
+  queue_as :default
+
+  def perform
+    results = { total: 0, processed: 0, errors: [] }
+    queue_url = ENV.fetch('SQS_OUTPUT_QUEUE_URL')
+    Rails.logger.info("Reading messages from queue #{queue_url}...")
+
+    begin
+      # https://docs.aws.amazon.com/sdk-for-ruby/v3/developer-guide/sqs-example-poll-messages.html
+      # Poller retrieves messages until there are none left and deletes them as it goes
+      poller = Aws::SQS::QueuePoller.new(queue_url)
+
+      poller.poll(max_number_of_messages: 10, wait_time_seconds: 10, idle_timeout: 0.5) do |messages|
+        messages.each do |msg|
```

Comment:
My preference would be to break down the logic in `perform` into separate methods within the same class, so as to avoid queueing more jobs than is necessary. I'd welcome @JPrevost's thoughts on this, since he authored #808 and co-authored this one.